Viktoriia_Kovalenko_JavaRush_TextQuest

cladyland opened the pull request.

cladyland committed:
+ Sentence, Answer, Question and Card classes created;
+ pom.xml and cards.yml created

cladyland committed:
+ enum Status created;
+ Answer.java: added a new variable;
+ cards.yml edited

cladyland committed:
Card.java, Sentence.java (with all subclasses): added default constructor

cladyland committed:
some classes moved to a new package .basis;
cards.yml edited;
+ Defeat.java and defeats.yml created

cladyland committed:
defeats.yml edited;
+ CardsManager.java created

cladyland committed:
CardsManager.java: deserializeYamlFile method changed

cladyland committed:
+ AppContextListener.java, AuthenticationFilter.java, NewNickNameFilter.java, Player.java, PlayerRepository.java created

cladyland committed:
NewNickNameFilter.java: added methods to check the nickname for invalid characters;
+ index.jsp, web.xml created

cladyland committed:
+ PathsJsp.java, quest.jsp created;
RegisterServlet.java: doPost() changed, added a check if the nickname is free for registration
NewNickNameFilter.java: changed name of a variable

cladyland committed:
CardsManager.java: the implementation of the class logic has been changed;
AppContextListener.java: added new attributes to servlet context

cladyland committed:
AuthenticationFilter.java: added additional verification for new user and redirection for registered one;
RegisterServlet.java: set an attribute to current session

cladyland committed:
pom.xml edited

cladyland committed:
CardsManager.java: added new variables, deserializeYamlFile() edited;
PathsJsp.java: getPath() changed to toString();
RegisterServlet.java: added attributes to httpSession

cladyland committed:
NewNickNameFilter.java: added a check if a user entered his name;
RegisterServlet.java: deleted overriding of doGet(), nickNameIsFree() refactored;
Sentence.java (with all subclasses): changed type of "id" variable

cladyland committed:
Card.java: added a new variable, changed type of "answers" variable;
cards.yml edited

cladyland committed:
Player.java: added new variables and a method;
PathsJsp.java: added new path;
RegisterServlet.java: remove one attribute;
+ start.jsp created;
+ LogicFilter && LogicServlet created

cladyland committed:
quest.jsp edited;
LogicServlet.java: the bug due to which the information is not displayed in the browser has been fixed

cladyland committed:
AppContextListener.java: added new attribute;
PathsJsp.java: added a new variable;
LogicServlet.java: searching for questions and answers and transferring the result to .jsp files is implemented

cladyland committed:
LogicServlet.java: GET response bug fixed;
Status.java: new variable added

cladyland committed:
Player.java: added methods to change the player's statistic

cladyland committed:
LogicServlet.java: added method for finding defeats messages, doPost() edited

cladyland committed:
+ PlayerService.java created;
RegisterServlet.java: working with the repository is replaced by working with the service;
PlayerRepository.java: added check for null;
AppContextListener.java: added new variable;

all classes that were outside the main package have been moved to kovalenko.vika package

cladyland committed:
AppContextListener.java: added new attributes, removed some old ones;
PlayerService.java: added new method;
+ QuestService && QuestDefaultException created;
LogicServlet.java: the logic of the servlet has been changed

cladyland committed:
Card.java, Sentence.java (with all subclasses): creation of constructors is replaced by annotations

cladyland committed:
added null checks and @NonNull annotation for base classes variables

cladyland committed:
Player.java: added new method and checking if player statistic can be changed;
+ PlayerTest.java created

cladyland committed:
+ PlayerSettingsException created

cladyland committed:
+ PlayerRepositoryTest, PlayerServiceTest created;
PlayerTest.java: added new tests;
PlayerRepository.java: edited method, created constructor;
PlayerService.java: added annotation;
Player.java: filling out statistics is carried out in a separate method;
pom.xml edited

cladyland committed:
PlayerService.java: added null check;
PlayerServiceTest.java: new tests created

cladyland committed:
QuestService.java: added null check;
+ QuestServiceTest.java created

cladyland committed:
QuestServiceTest.java: added new variables and tests;
+ AuthenticationFilterTest.java created

cladyland committed:
NewNickNameFilter.java: exception message edited;
+ LogicFilterTest and NewNickNameFilterTest created

cladyland committed:
added @ExtendWith to some *Test classes and also edited those classes;
+ RegisterServletTest created

cladyland committed:
PlayerService.java: added new method;
RegisterServlet.java: private methods are replaced by calling methods of the corresponding service;
RegisterServletTest.java: added new tests

cladyland committed:
PlayerServiceTest.java: added annotation @ExtendWith;
LogicServlet.java: added ServletConfig, edited doPost() method;
+ LogicServletTest created

cladyland committed:
CardsManager.java, PathsJsp.java and PlayerRepository.java moved to new package

cladyland committed:
imports edited

cladyland committed:
Answer, Card and Sentence: logger added;
pom.xml edited

cladyland committed:
Player and PlayerRepository: logger added;
CardsManager.java: additional checks and logger added

cladyland committed:
logs added, bugs fixed;
+ log4j configuration file created

cladyland committed:
index.jsp edited;
+ style.css created;
+ background picture added

cladyland committed:
index.jsp, start.jsp and style.css edited

cladyland committed:
quest.jsp, start.jsp and style.css edited;
+ end.jsp added

cladyland committed:
LogicFilter, LogicServlet, RegisterServletTest edited;
web.xml edited;
+ error.jsp created

cladyland committed:
LogicServletTest.java, pom.xml, .gitignore edited;
war file created

cladyland committed:
README.md edited

akupreychik commented on src/main/java/kovalenko/vika/AppContextListener.java:
```diff
@@ -0,0 +1,40 @@
+    public void contextInitialized(ServletContextEvent contextEvent) {
+        LOG.info("'Servlet context' initialization begins...");
+
+        var playerRepository = new PlayerRepository();
```
like!

akupreychik commented on src/main/java/kovalenko/vika/AppContextListener.java:
```diff
@@ -0,0 +1,40 @@
+        var playerRepository = new PlayerRepository();
+        var playerService = new PlayerService(playerRepository);
+        var cardsManager = new CardsManager();
+        cardsManager.createCards("cards.yml");
```
а текст с названием файла в константу

akupreychik commented on src/main/java/kovalenko/vika/LogicServlet.java:
```diff
@@ -0,0 +1,96 @@
+
+        Card playerCard = questService.getCurrentCard(playerCardId);
+
+        if (nonNull(answerParam)) {
+            int playerAnswerId = Integer.parseInt(answerParam);
+            Status playerStatus = questService.getPlayerAnswerStatus(playerCardId, playerAnswerId);
+
+            if (playerStatus == Status.NEXT) {
+                playerCard = questService.getNextCard(playerCardId);
+                session.setAttribute("cardID", playerCard.getId());
+            } else {
+                if (playerStatus == Status.DEFEAT) {
+                    String defeatMessage = questService.getDefeatMessage(playerAnswerId) + "\nYOU LOSE";
+                    req.setAttribute("defeat", defeatMessage);
+                } else if (playerStatus == Status.VICTORY) {
+                    String winMessage = questService.getVictoryMessage() + "\nYOU WIN";
+                    req.setAttribute("victory", winMessage);
+                }
+                req.setAttribute("statistic", playerService.setAndGetPlayerStatistic(player, playerStatus));
+                session.removeAttribute("cardID");
+
+                forwardTo(END_JSP, req, resp);
+                return;
+            }
+        }
```
лучше разбить на маленькие методы. сейчас это сложновато читать

akupreychik commented on src/main/java/kovalenko/vika/db/PathsJsp.java:
```diff
@@ -0,0 +1,20 @@
+package kovalenko.vika.db;
+
+public enum PathsJsp {
```
используй lombok

akupreychik reviewed (approved):
Отличный проект! Тесты классные + не забудь залить на render

AndryLat commented on pom.xml:
```diff
@@ -0,0 +1,112 @@
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Пустая строка в конце файла для гита обязательна.

AndryLat commented on src/main/java/kovalenko/vika/AuthenticationFilter.java:
```diff
@@ -0,0 +1,52 @@
+import static kovalenko.vika.db.PathsJsp.INDEX_JSP;
+
+@WebFilter(filterName = "AuthenticationFilter", value = "/")
+public class AuthenticationFilter implements Filter {
```
Огонь!

AndryLat commented on src/main/java/kovalenko/vika/RegisterServlet.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Чем аргументирован модификатор протектед ?

AndryLat commented on src/main/java/kovalenko/vika/RegisterServlet.java:
```diff
@@ -0,0 +1,59 @@
+            doGet(req, resp);
+            return;
+        }
+
```
не вижу повсеместного использования логгера.

AndryLat commented on src/main/java/kovalenko/vika/db/PlayerRepository.java:
```diff
@@ -0,0 +1,33 @@
+public class PlayerRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(PlayerRepository.class);
+
+    @Getter
```
Я рассказывал, что вот так просто брать и получать полноценную мапу, чтобы потом ее можно было менять кому угодно, а не конкретно этому классу - плохо. Всегда возвращаем копию, если так сильно надо, а в идеале мы вообще не отдаем доступа к этому хранилищу, даем только методы для работы с ним.

AndryLat reviewed (approved):
Топовый проектик!

cladyland committed:
AppContextListener: file names with settings are assigned to constants;
LogicServlet: part of the logic is transferred to a private method;
PathsJsp: added lombok annotation;
PlayerRepository: custom getter created;
pom.xml and text-quest-1.0.war updated
